machine/v1alpha1/types_openstack.go

```diff
@@ -65,6 +65,7 @@ type OpenstackProviderSpec struct {
 	FloatingIP string `json:"floatingIP,omitempty"`
 
 	// The availability zone from which to launch the server.
+	// It is an error to set both this and `ComputeAvailabilityZone` in `FailureDomain`.
```

Comment:
Can we add a CEL validation that prevents this? (Edit: No, this is unstructured)